cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/HttpHelper.java

```diff
@@ -0,0 +1,50 @@
+package zone.cogni.libs.sparqlservice.impl;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.fluent.Response;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.BasicAuthCache;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class HttpHelper {
+
+  public static HttpClientContext createAuthenticationHttpContext(String user, String password) {
+    CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+    credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(user, password));
+
+    HttpClientContext context = HttpClientContext.create();
+    context.setCredentialsProvider(credentialsProvider);
+    context.setAuthCache(new BasicAuthCache()); //so after first call it will know it has to send the authentication
+    return context;
+  }
+
+  public static void checkAndDiscardResponse(Response response) throws IOException {
+    try {
+      HttpResponse httpResponse = response.returnResponse();
+      StatusLine statusLine = httpResponse.getStatusLine();
+      if (statusLine.getStatusCode() / 100 == 2) return;
+      throw new RuntimeException("Upload didn't answer 2xx code " + statusLine + ": " + getBody(httpResponse));
```

Comment:
I guess different backends return different error messages... do we need to add a translation in SparqlService? now it is basically, take response and dump it (which probably is good enough for exception messages)

Reply:
We can do this in the context of an in depth redesign of the sparql services. 
Here it's like we do for all the other SparqlService: 2xx code => success, any other code: fail and log why